src/domain/flow-system/domain/src/aggregates/flow/flow.rs

```diff
@@ -80,6 +80,21 @@ impl Flow {
         }
     }
 
+    /// Indicate flow in enqueued for activation at particular time
+    pub fn enqueue(
+        &mut self,
+        now: DateTime<Utc>,
+        enqueued_for: DateTime<Utc>,
```

Comment:
Very minor: When I first read `enqueued_for` - it sounded like a duration to me.

I really liked how you described it as "activation" in the above comment, so mb `activation_time` would be a better name for this field.

My mental model is still that we are not "queueing" flows - we are determining the appropriate time when they should run. Queue-based logic is for tasks. So perhaps event could be named `Scheduled`. Thus we will be _**scheduling** a flow for **activation** at time T_.

Reply:
Did a big rename over the system to completely destroy term "enqueue" within the flow system.